lib/js/src/rpc/messages/RegisterAppInterface.js

```diff
@@ -0,0 +1,381 @@
+* IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+* ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+* LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+* CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+* SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+* INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+* CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+* ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+* POSSIBILITY OF SUCH DAMAGE.
+*/
+
+import { RpcRequest } from '../RpcRequest.js';
+import { SdlMsgVersion } from '../structs/SdlMsgVersion.js';
+import { TTSChunk } from '../structs/TtsChunk.js';
+import { DeviceInfo } from '../structs/DeviceInfo.js';
+import { AppInfo } from '../structs/AppInfo.js';
+import { TemplateColorScheme } from '../structs/TemplateColorScheme.js';
+
+import { Language } from '../enums/Language.js';
+import { AppHMIType } from '../enums/AppHMIType.js';
+import { FunctionID } from '../enums/FunctionID.js';
+
+class RegisterAppInterface extends RpcRequest {
+
+    constructor(store) {
+        super(store);
+        this.setFunctionName(FunctionID.REGISTER_APP_INTERFACE);
+    }
+
+    /**
+    * @param {SdlMsgVersion} The max RPC Spec version supported by this library
+    * @return {RegisterAppInterface}
+    */
+    setSdlMsgVersion(sdlMsgVersion) {
+        this.validateType(SdlMsgVersion, sdlMsgVersion);
+
+        this.setParameter(RegisterAppInterface.KEY_SDL_MSG_VERSION, sdlMsgVersion);
+        return this;
+    }
+
+    /**
+    * @return {SdlMsgVersion}
+    */
+    getSdlMsgVersion() {
+        return this.getParameter(RegisterAppInterface.KEY_SDL_MSG_VERSION);
+    }
+
+
+    /**
+    * @param {String} appName the name of the app that is registering
+    * @return {RegisterAppInterface}
+    */
+    setAppName(appName) {
+        //TODO do we need this? this.validateType(String, appName);
+
+        this.setParameter(RegisterAppInterface.KEY_APP_NAME, appName);
+        return this;
+    }
+
+    /**
+    * @return {String} the app name
+    */
+    getAppName() {
+        return this.getParameter(RegisterAppInterface.KEY_APP_NAME);
+    }
+
+    /**
+   * @param {Array<TTSChunk>} ttsNames TTS string for VR recognition of the mobile application name, e.g. "My S D L App". 
+   *                                   Meant to overcome any failing on speech engine in properly pronouncing / understanding 
+   *                                   app name. Needs to be unique over all applications from the same device. May not be 
+   *                                   empty. May not start with a new line character. Only characters from char set 
+   * @return {RegisterAppInterface}
+   */
+    setTtsName(ttsNames) {
+        this.setParameter(RegisterAppInterface.KEY_TTS_NAME, ttsNames);
```

Comment:
I believe there should be added `this.validateType(TTSChunk, name);`, right?

Reply:
There is currently no way to validate an Array of values in that method.